app/services/doc_auth_router.rb

```diff
@@ -199,6 +199,17 @@ def self.client(warn_notifier: nil)
   # rubocop:enable Layout/LineLength
 
   def self.doc_auth_vendor
+    if IdentityConfig.store.doc_auth_vendor_randomize
+      target_percent = IdentityConfig.store.doc_auth_vendor_randomize_percent
+
+      target_percent = target_percent > 100 ? 100 : target_percent
+      target_percent = target_percent < 0 ? 0 : target_percent
+
+      if rand(100) < target_percent
+        return IdentityConfig.store.doc_auth_vendor_randomize_alternate_vendor
+      end
+    end
```

Comment:
Something like...

```suggestion
  def self.doc_auth_vendor(user)
    if IdentityConfig.store.doc_auth_vendor_randomize
      target_percent = IdentityConfig.store.doc_auth_vendor_randomize_percent

      target_percent = target_percent > 100 ? 100 : target_percent
      target_percent = target_percent < 0 ? 0 : target_percent

      max_sha = (16**64)-1
      user_value = Digest::SHA256.hexdigest(user.uuid).to_i(16).to_f / max_sha * 100

      if user_value < target_percent
        return IdentityConfig.store.doc_auth_vendor_randomize_alternate_vendor
      end
    end
```

Reply:
Herm, I had forgotten that the doc_auth_vendor method was used a few places.

This solution would not randomize by request, if I understand it correctly. Each user would deterministically always be a particular number between 0 and 100 and their vendor would only change when the percentage was changed.

Am I understanding correctly?

Reply:
yes this would randomize by user, because if we did it any other way, two calls inside the same request might be different

Reply:
Yeah, you're right. Though I think I'd change over to using the session uuid rather than the user uuid. 

Since that was one of the discussed ways of breaking up users.

Reply:
After digging more it turns out that I thought there was a uuid directly associated with the session but there isn't really. So I kept looking around and found the `unique_session_id` associated with the user. That seems promising so I'm investigating using that a way to make vendor deterministic by session rather than user.

Reply:
Is that session ID kept the same across the hybrid flow? In general, I'm wary of our session data, and I really think that user ID is the most stable identifier we have, is there a good reason not to use it?

Reply:
Yeah, I think the user id being more stable is preferable in this case

Reply:
Yeah, after lots of digging and looking for other options it basically has to be by user. Though I chose `document_capture_session_uuid` as it was easily available in all of the places I needed to call it. But that is consistent for the purposes of randomization of the vendor since the user has a single document capture session stored for them.

Reply:
> Yeah, after lots of digging and looking for other options it basically has to be by user

We can easily hop to `document_capture_session.user`?

Reply:
That is possible to do but the `dcs_uuid` accomplishes the same goal with anything additional. There is a single `dcs_uuid` created per user, and even if it is reset for some reason it would not be reset in the middle of an existing document capture session.

But it's always possible I'm missing something.